src/main/java/pl/zimi/zadanka/Zad4.java

```diff
@@ -0,0 +1,94 @@
+package pl.zimi.zadanka;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/*
+source
+https://leetcode.com/problems/reverse-words-in-a-string/
+ */
+public class Zad4 {
+    public static String reverseString(String s) {
+
+        String[] splitedS = s.split(" ");
+        String sReverse = "";
+        boolean flaga = false;
+        for (int i = splitedS.length - 1; i >= 0; i--) {
+
+            if (splitedS[i].length() > 0) {
+//            if (true) {     //ten zostawia artefakty na koncu linii
+                if (flaga == true) {
+                    sReverse += " " + splitedS[i];
+                } else {
+                    sReverse += splitedS[i];
+                    flaga = true;
+                }
+            }
+        }
+        return sReverse;
+    }
+
+    public static String reverseStringPROBLEMY(String s) {
+        List<String> myAlist = new ArrayList<String>();
+        myAlist.add("Geeks");
+        myAlist.add("Practice");
+        myAlist.add("Quiz");
+        System.out.println("Original ArrayList : " + myAlist);
+        myAlist.remove("Quiz");
+        System.out.println("Modified ArrayList : " + myAlist);
+
+
+        String sReverse = new String();
```

Comment:
I detect that this code is problematic. According to the [Performance (PERFORMANCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#performance-performance), [Dm: Method invokes inefficient new String() constructor (DM_STRING_VOID_CTOR)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-method-invokes-inefficient-new-string-constructor-dm-string-void-ctor).
Creating a new java.lang.String object using the no-argument constructor wastes memory because the object so created will be functionally indistinguishable from the empty string constant "".  Java guarantees that identical string constants will be represented by the same String object.  Therefore, you should just use the empty string constant directly.

Reply:
Ok, I will change that